src/timeseriesflattener/flattened_dataset.py

```diff
@@ -303,6 +304,24 @@ def _drop_records_outside_interval_days(
             axis=1,
         )
 
+    @staticmethod
+    def _embed_text_values(
+        df: pd.DataFrame,
+        embedding_fn: Callable,
+        dim_reduction_fn: Optional[Callable],
+    ) -> pd.DataFrame:
+        """Embeds text values using the embedding_fn and optionally reduces the
+        dimensionality using dim_reduction_fn. Assumes that the text is stored
+        in the column "value"."""
+        embedding = embedding_fn(df["value"])
+        if dim_reduction_fn:
+            embedding = dim_reduction_fn(embedding)
+
+        df = df.drop("value", axis=1)
+        # make multiindex with embedding as 'value'
+        df = pd.concat([df, embedding], axis=1, keys=["df", "value"])
+        return df
```

Comment:
Since this is a static method, maybe it should be moved to a separate module outside of timeseriesflattener?

This goes for quite a few of the methods, but this is distinctly different from flattening of timeseries, so an easy candidate to start with :-)